back/spec/acceptance/home_pages_spec.rb

```diff
@@ -0,0 +1,82 @@
+  context 'when admin' do
+    before do
+      @admin = create(:admin)
+      token = Knock::AuthToken.new(payload: @admin.to_token_payload).token
+      header 'Authorization', "Bearer #{token}"
+    end
+
+    patch 'web_api/v1/home_page' do
+      with_options scope: :home_page do
+        parameter :banner_enabled, 'if the banner is enabled'
+        # parameter :banner_layout, 'if the banner is enabled'
+        # parameter :banner_avatars_enabled, 'if the banner is enabled'
+        # parameter :banner_signed_in_header_multiloc, 'if the banner is enabled'
+        # parameter :banner_signed_out_header_multiloc, 'if the banner is enabled'
+        # parameter :banner_signed_out_subheader_multiloc, 'if the banner is enabled'
+        # parameter :banner_signed_out_header_overlay_color, 'if the banner is enabled'
+        # parameter :banner_signed_out_header_overlay_opacity, 'if the banner is enabled'
+        # parameter :cta_signed_in_text_multiloc, 'if the banner is enabled'
+        # parameter :cta_signed_in_type, 'if the banner is enabled'
+        # parameter :cta_signed_in_url, 'if the banner is enabled'
+        # parameter :cta_signed_out_text_multiloc, 'if the banner is enabled'
+        # parameter :cta_signed_out_type, 'if the banner is enabled'
+        # parameter :cta_signed_out_url, 'if the banner is enabled'
+        # parameter :header_bg, 'if the banner is enabled'
+        # parameter :top_info_section_enabled, 'if the banner is enabled'
+        # parameter :top_info_section_multiloc, 'if the banner is enabled'
+        # parameter :bottom_info_section_enabled, 'if the banner is enabled'
+        # parameter :bottom_info_section_multiloc, 'if the banner is enabled'
+        parameter :events_enabled, 'if the events are enabled'
+        # parameter :projects_enabled, 'if the banner is enabled'
+        # parameter :projects_header_multiloc, 'if the banner is enabled'
+        parameter :pinned_admin_publication_ids, 'the IDs of admin publications that are pinned to the page', type: :array
+      end
+      ValidationErrorHelper.new.error_fields(self, HomePage)
+
+      let(:banner_enabled) { true }
+      let(:events_enabled) { true }
+
+      example_request 'Update the current home page' do
+        expect(response_status).to eq 200
+        json_response = json_parse(response_body)
+        expect(json_response.dig(:data, :attributes, :banner_enabled)).to be true
+      end
+
+      describe do
+        let(:project_one) { create(:project) }
+        let(:project_two) { create(:project) }
+        let(:pinned_admin_publication_ids) do
+          [project_one.admin_publication.id, project_two.admin_publication.id]
+        end
+
+        example_request 'Update pins to a page' do
+          json_response = json_parse(response_body)
+          expect(response_status).to eq 200
+          expect(json_response.dig(:data, :relationships, :pinned_admin_publications, :data).length).to eq(2)
+        end
+      end
+    end
```

Comment:
There needs to be at least one more test for removing a pin from the page, but see my comment above, not sure how to do that with the nested attributes.

@nTraum as you mentioned I think we also need tests on the edge cases, to make sure trying to add more than 3 pages fails, etc.

Reply:
If I understand right:

For the first test we're adding the two projects (via admin publication ids) to a HomePage via the request. In the second test we're setting 3 projects in the code before the test, and then we make the request, which overwrites the in-code association with the two IDs we're sending with the request, and so the length should be 2 at the end?

Reply:
That is correct Lucas! :)